spring-cloud-kubernetes-test-support/src/main/java/org/springframework/cloud/kubernetes/integration/tests/commons/K8SUtils.java

```diff
@@ -172,11 +173,20 @@ public void waitForDeployment(String deploymentName, String namespace) {
 
 	public void waitForDeploymentToBeDeleted(String deploymentName, String namespace) {
 		await().timeout(
-				Duration.ofSeconds(90)).until(
-						() -> appsApi
-								.listNamespacedDeployment(namespace, null, null, null,
-										"metadata.name=" + deploymentName, null, null, null, null, null)
-								.getItems().isEmpty());
+			Duration.ofSeconds(90)).until(
+			() -> {
+				try {
+					appsApi
```

Comment:
What is the advantage of doing it this way?

Reply:
the original setup flow is working like "deploy" -> "wait" -> "deploy" -> "wait" ->..., this pull makes "deploy" and "wait" work in parallel, wdyt?

Reply:
I mean the change from `listNamespaceDeployment` to `readNamespacedDeplayment`, what is the advantage?

Reply:
kube-apiserver will be loading all deployments from etcd into memory and filtering w/ the given selector `metata.name` which has a time cost of `O(# of list items)` upon `listNamespaceDeployment`, while `readNamespacedDeplayment` will only load single object. the difference also applies to `deleteCollection` and single `delete`.